### DISCUSSION TITLE
How to address migrations from different versions of dev branches?

### DISCUSSION THREAD

**Discussion by amanangira** [golang-migrate/migrate#614] [Q&A]
As far as I understand, this package holds only the current migration in the configured `migrations` table. 

Problem 

1. Consider branch `dev_a` and `dev_b` branched out from the latest `master`.
2. Some migrations were added to both `dev_a` and `dev_b`.
3. Assume developer X working on `dev_a` switched to `dev_b`. Now, as soon as he runs the migrations he/she is given the error `Error occurred running migrations: no migration found for version XXXXXXX: read down for version XXXXXXX : file does not exist`. Now in order to resolve it he/she has to do it manually figuring out the right version.

How do I handle this scenario? Is there a better way around this problem that I am overlooking? I have previously worked with Symfony (PHP framework) that puts in all the versions in the configured `migrations` table that has been executed. And simply shows a warning for migrations that are not found in the codebase. We are thinking of implementing something similar in case there is no way around this problem using the provided configurations in this package.

**Comment by dhui** [comment]
Currently, `migrate` doesn't handle simultaneous development of migrations. Here's a [proposal](https://github.com/golang-migrate/migrate/issues/470) for solving this problem. Any feedback you can make in that issue would be great!

In your example, if you don't have many migrations (or they don't take long to run), I'd start with a clean DB and run all migrations when switching branches. If that's not feasible, I'd find the migrations that need to be reverted and applied by running `git log --name-status master..dev_b ${MY_MIGRATIONS_DIR}` when switching branches (from `dev_a` to `dev_b`).
